packages/legend-application-query/src/stores/QueryEditorStore.ts

```diff
@@ -638,6 +674,17 @@ export class MappingQueryCreatorStore extends QueryEditorStore {
       },
     };
   }
+
+  async getExportConfigurationDirectSave(): Promise<QueryExportConfiguration> {
+    return {
+      decorator: (query: Query): void => {
```

Comment:
do we have to duplicate the code everywhere? Why don't we pass a boolean to getExportConfiguartion so that we kno if it is a direct save or a new query